### PR TITLE
Live view refactor

### LINKED ISSUE #10: Move html to templates
Move HTML from `live/console_live.ex` to templates

### PR DESCRIPTION
**Pull request by jmbejar**
This PR extract part of the main live view to _live view components_, so responsibilities are better divided.

We have now three additional components: `CommandInputComponent`, `HistoryComponent`, `SidebarComponent`, which are orchestrated from the already existing `ConsoleLive` module. Each one takes care of rendering and handing the events triggered in a portion of the UI.

It fixes #10

### CODE REVIEW

File: lib/elixir_console_web/live/console_live/command_input_component.ex [hunk 1]
```diff
@@ -0,0 +1,100 @@
+defmodule ElixirConsoleWeb.ConsoleLive.CommandInputComponent do
+  @moduledoc """
+  Live component module that implements the portion of the application where
+  users prompt the commands. It also handle special key events to mimic the iEX
```
Comment (nicolasferraro) [inline lib/elixir_console_web/live/console_live/command_input_component.ex]: `handle` -> `handles`

File: lib/elixir_console_web/live/console_live/command_input_component.ex [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+     assign(
+       socket,
+       history_counter: 0,
+       input_value: "",
+       caret_position: 0
+     )}
+  end
+
+  # TAB KEY
+  def handle_event("suggest", %{"keyCode" => 9, "value" => value}, socket) do
```
Comment (nicolasferraro) [inline lib/elixir_console_web/live/console_live/command_input_component.ex]: Would it make sense to define constants with the keycode values? Eg: `@tab_keycode 9`

Comment (nicolasferraro) [reply lib/elixir_console_web/live/console_live/command_input_component.ex]: That would save us the comment on the previous line too I guess

File: lib/elixir_console_web/live/console_live/command_input_component.ex [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+      suggestions ->
+        send(self(), {:update_suggestions, suggestions})
+        {:noreply, assign(socket, input_value: "")}
+    end
+  end
+
+  # KEY UP
+  def handle_event("suggest", %{"keyCode" => 38}, socket) do
+    counter = socket.assigns.history_counter
```
Comment (nicolasferraro) [inline lib/elixir_console_web/live/console_live/command_input_component.ex]: WDYT about using pattern matching here?
```elixir
%{history_counter: counter, history: history} = socket.assigns
```

File: lib/elixir_console_web/templates/console/history/index.html.leex [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div class="flex-1"></div>
+  <div class="p-2" id="commandOutput" phx-update="append">
+    <%= for output <- @output do %>
+      <div id="command<%= output.id %>" class="text-gray-300 font-medium"><%= print_prompt() %><%= format_command(output.command) %></div>
+      <div id="output<%= output.id %>" class="text-teal-300">
+        <%= if output.result do output.result end %>
```
Comment (nicolasferraro) [inline lib/elixir_console_web/templates/console/history/index.html.leex]: Question out of ignorance, what would happen if you just do `<%= output.result %>` and it's not defined? wouldn't it ignore it? (or render nothing to be more precise)

Comment (jmbejar) [reply lib/elixir_console_web/templates/console/history/index.html.leex]: That's a good question that I can not answer off the top of my head.
Let me try it and confirm.

File: lib/elixir_console_web/live/console_live/command_input_component.ex [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+        counter + 1 >= length(history) ->
+          {[List.last(history)], counter}
+      end
+
+    {:noreply, assign(socket, input_value: input_value, history_counter: new_counter)}
+  end
+
+  # KEY DOWN
+  def handle_event("suggest", %{"keyCode" => 40}, socket) do
```
Comment (iaguirre88) [inline lib/elixir_console_web/live/console_live/command_input_component.ex]: Same as Nico here about using `@key_down_keycode` (just for you not to forget :p)

File: lib/elixir_console_web/live/console_live/command_input_component.ex [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      cond do
+        history == [] ->
+          {[], 0}
+
+        counter > 0 ->
+          {[Enum.at(history, counter - 1)], counter - 1}
+
+        counter <= 0 ->
+          {[List.first(history)], 0}
+      end
```
Comment (iaguirre88) [inline lib/elixir_console_web/live/console_live/command_input_component.ex]: NIT: Maybe we can extract this to a private function like `previous_history_entry`

File: lib/elixir_console_web/live/console_live/command_input_component.ex [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      cond do
+        history == [] ->
+          {[], 0}
+
+        counter + 1 < length(history) ->
+          {[Enum.at(history, counter + 1)], counter + 1}
+
+        counter + 1 >= length(history) ->
+          {[List.last(history)], counter}
+      end
```
Comment (iaguirre88) [inline lib/elixir_console_web/live/console_live/command_input_component.ex]: NIT: Maybe we can extract this to a private function like `next_history_entry`

File: lib/elixir_console_web/templates/console/history/index.html.leex (hunk 4 above)
Comment (jmbejar) [reply lib/elixir_console_web/templates/console/history/index.html.leex]: You're totally right... nothing is rendering if the value is nil, so the conditional is useless

**Pull request merged**